0.2.0/docker-entrypoint.sh

```diff
@@ -0,0 +1,279 @@
+    done
+    
+    case "${JANUS_STORAGE_BACKEND}" in
+    "berkeleyje")
+        write_berkeley_props
+       ;;
+    "cassandra")
+        write_cassandra_props
+       ;;
+    "cql")
+       write_cql_props
+       ;;
+    "hbase")
+       write_hbase_props
+       ;;
+    esac
+
+}
+
+write_es_index_props() {
+
+  local index_name=${1}
+
+  for prop in \
+      client-only \
+      cluster-name \
+      health-request-timeout \
+      ignore-cluster-name \
+      interface \
+      load-default-node-settings \
+      local-mode \
+      sniff \
+      ttl-interval \
+    ; do
+
+      write_prop ${prop} 'index.'"${index_name}"'.elasticsearch.${prop}' 'JANUS_INDEX_'"${index_name^^}"'_ES_${prop}'
+  done
+
+}
+
+write_index_props() {
+
+  local index_name=${1}
+
+  for prop in \
+      backend \
+      conf-file \
+      directory \
+      hostname \
+      index-name \
+      map-name \
+      max-result-set-size \
+      port \
+    ; do
+      write_prop ${prop} 'index.'"${index_name}"'.${prop}' 'JANUS_INDEX_'"${index_name^^}"'_${prop}'
+  done
+
+  backend="JANUS_INDEX_${index_name^^}_BACKEND"
+  case "${!backend}" in
+  "elasticsearch")
+      write_es_index_props ${index_name}
+     ;;
+  esac
+
+}
+
+writer_server_props() {
+
+  for prop in \
+      host \
+      port \
+      threadPoolWorker \
+      gremlinPool \
+      scriptEvaluationTimeout \
+      serializedResponseTimeout \
+      channelizer \
+      threadPoolBoss \
+      maxInitialLineLength \
+      maxHeaderSize \
+      maxChunkSize \
+      maxContentLength \
+      maxAccumulationBufferComponents \
+      resultIterationBatchSize \
+      writeBufferHighWaterMark \
+      writeBufferHighWaterMark \
+    ; do
+      # generate env variable for prop
+      local var="JANUS_GREMLIN_${prop^^}"
+      # get value of environment variable
+      local val="${!var}"
+      # if there is a value replace it in the properties file
+      if [ "$val" ]; then
+        if grep -q -E '^'"${prop}"'.*' ${GREMLIN_YAML}; then
+          sed -ri 's/^('"$prop"':).*/\1 '"$val"'/' ${GREMLIN_YAML}
+        else
+          echo "${prop}: ${val}" >> ${GREMLIN_YAML}
+        fi
+      fi
+    done
+}
+
+# running as root; step down to run as janusgraph user
+if [ "$1" == 'janusgraph' ] && [ "$(id -u)" == "0" ]; then
+    mkdir -p ${JANUS_DATA_DIR} ${JANUS_CONFIG_DIR}
+    chown -R janusgraph:janusgraph ${JANUS_DATA_DIR} ${JANUS_CONFIG_DIR}
+    chmod 700 ${JANUS_DATA_DIR} ${JANUS_CONFIG_DIR}
+
+    exec chroot --skip-chdir --userspec janusgraph:janusgraph / "${BASH_SOURCE}" "$@"
+fi
+
+# https://gist.github.com/samizuh/a978bc530e88847c2b5d8f75cb8caee0
+
+# running as non root user
+if [ "$1" == 'janusgraph' ]; then
+
+    mkdir -p ${JANUS_DATA_DIR}
+    chown -R "$(id -u):$(id -g)" ${JANUS_DATA_DIR} ${JANUS_CONFIG_DIR}
+    chmod 700 ${JANUS_DATA_DIR} ${JANUS_CONFIG_DIR}
+    chmod -R 600 ${JANUS_CONFIG_DIR}/*
+
+    writer_server_props
+    write_storage_props
+
```

Comment:
How about settings like `schema.*`, `query.*`, and `cache.*`? Aren't they missing?